torch/csrc/distributed/autograd/context/thread_local_context.cpp

```diff
@@ -0,0 +1,28 @@
+#include <torch/csrc/distributed/autograd/context/thread_local_context.h>
+
+namespace torch {
+namespace distributed {
+namespace autograd {
+
+namespace {
+thread_local ContextPtr::weak_type tl_context_weak_ptr;
+} // namespace
+
+ThreadLocalDistAutogradContext::ThreadLocalDistAutogradContext()
+    : prev_context_weak_ptr_(tl_context_weak_ptr),
+      context_weak_ptr_(tl_context_weak_ptr) {}
```

Comment:
Don't think we need this constructor if there is a static method which returns the thread local `tl_context_weak_ptr`.